daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala

```diff
@@ -290,9 +288,13 @@ object SValue {
         STextMap(map.mapValue(fromValue).toHashMap)
 
       case V.ValueGenMap(entries) =>
-        SGenMap(InsertOrdMap(entries.toSeq.map {
-          case (k, v) => SGenMap.Key(fromValue(k)) -> fromValue(v)
-        }: _*))
+        SGenMap(
+          entries
+            .foldLeft(TreeMap.newBuilder[SValue, SValue](svalue.Ordering)) {
+              case (b, (k, v)) => b += (fromValue(k) -> fromValue(v))
+            }
+            .result()
+        )
```

Comment:
Similarly use the `iterator map to` pattern.

Reply:
done